File: tests/fixtures/compare/sample.clj
;; Realistic Clojure code with various issues
(ns sample.core)

;; TODO: clean up debug output
(println "loading module")

(defn process [items]
  (println "processing" (count items))
  (defn helper [x]
    (Thread/sleep 100)
    (* x 2))
  (map helper items))

(defn with-reflection [obj]
  (.getClass obj))

(defn empty-fn [])

File: tests/fixtures/compare/sample.ex
# Realistic Elixir code with various issues
defmodule Sample do
  # TODO: remove debug calls

  def process(data) do
    IO.inspect(data, label: "debug")
    IO.puts("processing...")

    result = data
    |> Enum.map(&(&1 * 2))
    |> Enum.filter(&(&1 > 5))

    result
  end

  def empty_func do
  end

  def risky(x) do
    with {:ok, val} <- fetch(x) do
      raise "unexpected"
    end
  end

  defp fetch(_x), do: {:ok, 42}
end
